docs/source/v3-package-spec.rst

```diff
@@ -566,6 +566,16 @@ Filesystem path of source file.
   :Key: ``installPath``
   :Value: string
 
+Type: ``type``
+^^^^^^^^^^^^^^
+
+The ``type`` field declares the type of the source file. The field **should** be one
+of the following values: ``jsonabi``, ``solidity``, ``vyper``. 
+
+  :Required: No
+  :Key: ``type``
+  :Type: String
+
 ----
```

Comment:
A `type` field was added to `SourceObject` to make it simpler for compilers to identify what files they should be concerned about.